### DISCUSSION TITLE
Change in behavior to connect() function for Realtime

### DISCUSSION THREAD

**Discussion by jollygreenegiant** [supabase-community/supabase-kt#429] [Ideas]
@jan-tennert I'd like your thoughts on this before I make a PR - 

**The problem:**
When I connect to a realtime channel that is already connected, it throws an `IllegalStateException`, crashing my app.  This happens occasionally based on some lifecycle stuff on Android, where sometimes if you switch screens quickly, a channel may not get cleaned up entirely before trying to reconnect the next time you navigate to the screen.  I think there are ways to handle this client-side by checking if the channel is subscribed first, or wrapping it in a try/catch, but thought it might be better to handle it at the library level.

**Options:**
1. If you attempt to connect with a channel that is already connected, just return from the function and do nothing - if we want to connect to the channel, but it's already connected, then we're in the state we want to be in, no need to throw an error.

```
if (status.value == Realtime.Status.CONNECTED) {
    return
}
```

2. If you attempt to connect with a channel that is already connected, disconnect from the channel and reconnect to it.  I don't really know what the use case for this would be, but maybe it could be added as an option to the function?

```
if (status.value == Realtime.Status.CONNECTED) {
    if (reconnect) {
        // disconnect and reconnect
    } else {
        return
    }
}
```

Any thoughts on this?  Anything I'm not accounting for?

**Comment by jan-tennert** [comment]
Option 1 sounds good to me, I don't see any use case for option 2 as it would make more sense to just call disconnect and then connect again. If you make a PR, make sure to create it based of the `2.1.0` branch so we have it in the next release. Thank you!

**Reply by jollygreenegiant** [reply]
Made a PR for this - thank you!
